src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,179 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int length;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        if (isEmpty()) {
+            head = new Node<T>(null, value, null);
+            tail = head;
+            tail.setPrev(head);
+            head.setNext(head);
+            length++;
+            return;
+        }
+        Node<T> newTail = new Node<T>(tail, value, null);
+        tail.setNext(newTail);
+        newTail.setPrev(tail);
+        tail = newTail;
+        length++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == length || (isEmpty() && index > -1)) {
```

Comment:
```suggestion
        if (index == length) {
```
I think it could be simplified this way